akka-stream/src/main/resources/reference.conf

```diff
@@ -87,6 +87,21 @@ akka {
         # slightly more bytes than this limit (at most one element more). It can be set to 0
         # to disable the usage of the buffer.
         write-buffer-size = 16 KiB
+
+        # In addition to the buffering described for property write-buffer-size, more messages (frames)
+        # are collected if they arrive immediately before they are written as one buffer to the TCP output.
+        # This improves throughput for small messages. For example gRPC (HTTP/2) unary requests and responses
+        # consists of several frames (headers, body, trailers).
+        # By writing larger buffers the number of system calls can be reduced and thereby the write efficiency
+        # is improved.
+        # The frames are collected by inducing a short delay via a message round trip to the connection actor
+        # before flushing the write buffer. At most the number of round trips defined by this property,
+        # but as soon as no more frames arrive within such round trip the buffer is flushed.
+        # In other words, this adds a short latency. At low throughput rates the additional latency is
+        # very small, and it will increase at higher throughput but it is still small and bounded by
+        # the number of actor message round trips defined here.
+        # It can be set to 0 to disable this feature.
+        coalesce-writes = 10
```

Comment:
I did a bit of copy-editing here. The reference to gRPC is interesting for historic reasons (and may be added to the commit message) but might not be as interesting as documentation in the future. Feel free to change or keep the old etc.

```suggestion
        # In addition to the buffering described for property write-buffer-size, try to collect
        # more consecutive writes from the upstream stream producers.
        #
        # The rationale is to increase write efficiency by avoiding separate small 
        # writes to the network which is expensive to do. Merging those writes together
        # (up to `write-buffer-size`) improves throughput for small writes.
        #
        # The idea is that a running stream may produce multiple small writes consecutively
        # in one go without waiting for any external input. To probe the stream for
        # data, this features delays sending a write immediately by probing the stream
        # for more writes. This works by rescheduling the TCP connection stage via the
        # actor mailbox of the underlying actor. Thus, before the stage is reactivated
        # the upstream gets another opportunity to emit writes.
        #
        # When the stage is reactivated and if new writes are detected another round-trip
        # is scheduled. The loop repeats until either the number of round trips given in this
        # setting is reached, the buffer reaches `write-buffer-size`, or no new writes
        # were detected during the last round-trip.
        #
        # This mechanism ensures that a write is guaranteed to be sent when the remaining stream
        # becomes idle waiting for external signals.
        #
        # In most cases, the extra latency this mechanism introduces should be negligible,
        # but depending on the stream setup it may introduce a noticeable delay,
        # if the upstream continuously produces small amounts of writes in a
        # blocking (CPU-bound) way.
        #
        # In that case, the feature can either be disabled, or the producing CPU-bound
        # work can be taken off-stream to avoid excessive delays (e.g. using `mapAsync` instead of `map`).
        #
        # A value of 0 disables this feature.
        coalesce-writes = 10
```

Reply:
thanks, that's better